big_tests/tests/mod_event_pusher_rabbit_SUITE.erl

```diff
@@ -0,0 +1,321 @@
+
+end_per_group(_, Config) ->
+    Host = ct:get_config({hosts, mim, domain}),
+    dynamic_modules:stop("localhost", mod_event_pusher),
+    dynamic_modules:restore_modules(Host, Config),
+    escalus:delete_users(Config, escalus:get_users([bob, alice])).
+
+init_per_testcase(CaseName, Config0) ->
+    Config1 = escalus_fresh:create_users(Config0, [{bob, 1}, {alice, 1}]),
+    Config = Config1 ++ listen_to_rabbit(),
+    escalus:init_per_testcase(CaseName, Config).
+
+end_per_testcase(disconnected_users_delete_exchanges_when_unavailable = CaseName,
+                 Config) ->
+    escalus:end_per_testcase(CaseName, Config);
+end_per_testcase(CaseName, Config) ->
+    close_rabbit_connection(Config),
+    escalus:end_per_testcase(CaseName, Config).
+
+
+%%--------------------------------------------------------------------
+%% GROUP presence_status_publish
+%%--------------------------------------------------------------------
+
+connected_users_push_presence_events_when_change_status(Config) ->
+    escalus:story(
+      Config, [{bob, 1}, {alice, 1}],
+      fun(Bob, Alice) ->
+              %% GIVEN
+              escalus:wait_for_stanzas(Bob, 1),
+              escalus:wait_for_stanzas(Alice, 1),
+              BobJID = nick_to_jid(bob, Config),
+              AliceJID = nick_to_jid(alice, Config),
+              BobRoutingKey = presence_rk(BobJID),
+              AliceRoutingKey = presence_rk(AliceJID),
+              listen_to_presence_events_from_rabbit([BobJID, AliceJID], Config),
+              %% WHEN users generate some traffic.
+              send_presence_stanzas([Bob, Alice], 10),
+              %% THEN  wait for presence events from Rabbit.
+              ?assertReceivedMatch({#'basic.deliver'{
+                                       routing_key = BobRoutingKey},
+                                    #amqp_msg{}}, timer:seconds(5)),
+              ?assertReceivedMatch({#'basic.deliver'{
+                                       routing_key = AliceRoutingKey},
+                                    #amqp_msg{}}, timer:seconds(5))
+      end).
+
+connected_users_create_exchanges_when_available(Config) ->
+    escalus:story(
+      Config, [{bob, 1}, {alice, 1}],
+      fun(Bob, Alice) ->
+              %% GIVEN
+              Connection = proplists:get_value(rabbit_connection, Config),
+              Channel = proplists:get_value(rabbit_channel, Config),
+              BobJID = nick_to_jid(bob, Config),
+              AliceJID = nick_to_jid(alice, Config),
+              Exchanges = [BobJID, AliceJID],
+              %% WHEN users login
+              escalus:wait_for_stanzas(Bob, 1),
+              escalus:wait_for_stanzas(Alice, 1),
+              %% THEN exchanges are created
+              Result = [is_exchange_present(Connection, Channel, Exchange,
+                                 ?IF_EXCHANGE_EXISTS_RETRIES)
+                        || Exchange <- Exchanges],
+              ?assert(lists:all(fun(E) -> E == true end, Result))
```

Comment:
According to `is_exchange_present/4` it raises an exception if an error occurs. Therefore, you will never reach this line if error occurs (or exchange is not present).